### PR TITLE
tests: Add tests for affine

### PR DESCRIPTION
**Pull request by webmiche**
This PR adds some test case to get the affine dialect to 100%. This mainly consists of cases that throw exceptions.

### CODE REVIEW

File: tests/dialects/test_affine.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    assert e.value.args[
+        0] == "Expected BlockArguments to have the same types as the operands"
+
+
+def test_yield():
+    yield_ = Yield.get()
+    assert yield_.arguments == []
+
+
+test_yield()
```
Comment (math-fehr) [inline tests/dialects/test_affine.py:71]: Why do you have a call here?

**Review by math-fehr — CHANGES_REQUESTED** [review]
Just remove the toplevel call, and it's perfect!

File: tests/dialects/test_affine.py (hunk 1 above)
Comment (webmiche) [reply tests/dialects/test_affine.py]: I like to debug pytests like this :sweat_smile:

**Review by math-fehr — APPROVED** [review]
Perfect! Thanks!

**Review by superlopuh — COMMENTED** [review]
LGTM

**Pull request merged**